packages/odf/components/s3-browser/upload-objects/utils.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  humanizeBinaryBytes,
+  humanizeDecimalBytesPerSec,
+  humanizeMinutes,
+  humanizeSeconds,
+} from '@odf/shared/utils';
+import * as _ from 'lodash-es';
+import { UploadProgress, UploadStatus } from './types';
+
+export const getCompletedAndTotalUploadObjects = (objects: UploadProgress) => {
+  const totalObjects = Object.keys(objects).length;
+  const totalUploaded = Object.keys(objects).reduce((acc, curr) => {
+    if (objects[curr].total === objects[curr].loaded) {
+      acc += 1;
+      return acc;
+    }
+    return acc;
+  }, 0);
+  return [totalUploaded, totalObjects];
+};
+
+export const getUploadSpeed = (objects: UploadProgress): string => {
+  if (_.isEmpty(objects)) return '';
+  const files = Object.values(objects);
+  const uploadingFile = files.filter(
+    (item) => item.uploadState === UploadStatus.UPLOAD_START
+  )?.[0];
+  const totalTime = Date.now() - (uploadingFile?.startTime || Date.now());
+  const totalUploaded = uploadingFile?.loaded ?? 0;
+  const speed = humanizeDecimalBytesPerSec(totalUploaded / (totalTime / 1000));
+  return speed.string;
+};
+
+export const getTotalRemainingFilesAndSize = (
```

Comment:
"TotalRemaining" could include failed and the ones which have not started yet as well... hence name is bit confusing IMO...
```suggestion
export const getTotalUploadingFilesAndSize = (
```

Reply:
or if we wanna call it `getTotalRemainingFilesAndSize `, then include `INIT_STATE` ones as well ??